src/api/firebase.js

```diff
@@ -153,10 +154,11 @@ export async function updateItem(listId, itemId, checked) {
 	}
 }
 
-export async function deleteItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to delete an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function deleteItem(listId, itemId) {
+	const itemRef = doc(db, listId, itemId);
+	try {
+		return await deleteDoc(itemRef);
+	} catch (error) {
+		console.log('Delete item error', error);
```

Comment:
Were you able to trigger this error in testing?
I tried to simulate a network error by setting my Network conditions to offline (In chrome: Network tab > Network conditions) and nothing happens

Reply:
Hey @mentalcaries - we didn't go as far as that until now, but I tested your alternative here too, and the catch is not being triggered.

Would you prefer something as you suggested for the `updateItem`? Something like:

```
if (!itemRef) {

     console.log("Could not delete");
     return;

}

return await deleteDoc(itemRef);

```
Then we could delete the try-and-catch. 

Please let me know what you think and if that would be more helpful. And if not, what would you suggest in this case? Thank you! ✨

Reply:
If I delete an item offline, the item is removed from the list, but nothing changes in the database. Once I go back online, the database is immediately updated, no refresh needed. It made me think about a local cache that keeps a copy of the data. If I am reading the documentation correctly, this seems might be the case: "Cloud Firestore supports offline data persistence. This feature caches a copy of the Cloud Firestore data that your app is actively using, so your app can access the data when the device is offline. You can write, read, listen to, and query the cached data. When the device comes back online, Cloud Firestore synchronizes any local changes made by your app to the Cloud Firestore backend." Still exploring this right now ...

Reply:
https://firebase.google.com/docs/firestore/manage-data/enable-offline#get_offline_data

Reply:
Wondering now why last week I was getting an error while offline (this is when we were working on the validation for the addItem)

Reply:
@CorinaMurg good find. That's actually a really cool feature of Firebase esp if it's built in